packages/grid/_modules_/grid/utils/getTotalHeight.ts

```diff
@@ -5,17 +5,19 @@ import { ContainerProps, GridOptions } from '../models';
 export const getCurryTotalHeight = (
   internalOptions: GridOptions,
   containerSizes: ContainerProps | null,
+  headerRef: React.RefObject<HTMLDivElement>,
   footerRef: React.RefObject<HTMLDivElement>,
 ) => (size: any) => {
   const dataContainerHeight = (containerSizes && containerSizes.dataContainerSizes!.height) || 0;
   if (!internalOptions.autoHeight) {
     return size.height;
   }
   const footerHeight = (footerRef.current && footerRef.current.getBoundingClientRect().height) || 0;
+  const headerHeight = (headerRef.current && headerRef.current.getBoundingClientRect().height) || 0;
   let dataHeight = dataContainerHeight;
   if (dataHeight < internalOptions.rowHeight) {
     dataHeight = internalOptions.rowHeight * 2; // If we have no rows, we give the size of 2 rows to display the no rows overlay
   }
 
-  return footerHeight + dataHeight + internalOptions.headerHeight;
+  return headerHeight + footerHeight + dataHeight + internalOptions.headerHeight;
```

Comment:
```suggestion
  return headerHeight + footerHeight + dataHeight;
```

Reply:
What about we remove the `headerHeight` prop?

Reply:
> I think that we could have pushed the solution deeper. This is a partial solution. Consider the following case: https://codesandbox.io/s/cool-thunder-olrqk?file=/demo.js.
> 
> From my perspective, the best solution would be to move the `AutoSizer` to the grid zone.

I explored that solution as well. It can be done but it involves removing `outerStyle.height = 0;` and `outerStyle.width = 0;` from the `AutoSizer` to give the grid the correct dimensions. If I don't do it and I move the header and footer outside of the `AutoSizer` they collapse because the grid outer container has a height of 0. This btw maybe another bug that I am happy to solve but I don't know if anything will be impacting my removing those. When I tested it I didn't find any visible regressions. Maybe I can open another PR with the other solution and see if something will fail?